packages/datadog-plugin-dd-trace-api/src/index.js

```diff
@@ -0,0 +1,117 @@
+'use strict'
+
+const Plugin = require('../../dd-trace/src/plugins/plugin')
+const telemetryMetrics = require('../../dd-trace/src/telemetry/metrics')
+const apiMetrics = telemetryMetrics.manager.namespace('tracers')
+
+// api ==> here
+const objectMap = new WeakMap()
+
+module.exports = class DdTraceApiPlugin extends Plugin {
+  static get id () {
+    return 'dd-trace-api'
+  }
+
+  constructor (...args) {
+    super(...args)
+
+    const tracer = this._tracer
+
+    this.addSub('datadog-api:v1:tracerinit', ({ proxy }) => {
+      const proxyVal = proxy()
+      objectMap.set(proxyVal, tracer)
+      objectMap.set(proxyVal.appsec, tracer.appsec)
+      objectMap.set(proxyVal.dogstatsd, tracer.dogstatsd)
+    })
+
+    const handleEvent = (name) => {
+      const counter = apiMetrics.count('dd_trace_api.called', [
+        `name:${name.replace(':', '.')}`,
+        'api_version:v1',
+        `injection_enabled:${process.env.DD_INJECTION_ENABLED ? 'yes' : 'no'}`
```

Comment:
Can the `process.env.DD_INJECTION_ENABLED` check be hoisted to the outermost scope? I'm assuming it doesn't change throughout the lifetime of a process and it doesn't seem to be set during tests.

Also hopefully future config changes will make the config file easier to read in files like this one without needing to grab env vars everywhere.

Reply:
This particular env var is typically only used when config hasn't been initialized yet, so it's never been part of our config system IIRC. I'll hoist it as requested.